### ISSUE TITLE
KeyError ("time") for every package installation

### ISSUE THREAD

**Issue by liiyung** [JuliaLang/Pkg.jl#3521]
issue : out of nowhere, suddenly started getting the error `ERROR: KeyError: key "time" not found` when resolving package versions, regardless of the package I try to install

reproducibility : Not sure. I was able to install packages just fine before today; the last time I installed a package was a few days ago. The only thing I can think of is – today I tried using `@time` in JupyterLab, which I have not used previously.

versioninfo() : Julia Version 1.9.1, Julia Version 1.9.0 (tried both)
installation method : `brew install --cask julia` 
operating system : macOS Monterey 12.6, M1 2020

attempts at resolution : 
0. obligatory restart laptop
1. reinstalled Julia (which updated it from 1.9.0 to 1.9.1)
2. installed juliaup via `brew install juliaup`, and used juliaup to redownload/roll back to Julia 1.9.0 to try again.

The problem persists in both versions 1.9.0 and 1.9.1. As an example, I tried to redownload IJulia – since I already have this package, if successful, it should just inform me that there are no changes to Project.toml and Manifest.toml, as usual. The full error message from this is  
```
ERROR: KeyError: key "time" not found
Stacktrace:
  [1] getindex
    @ ./dict.jl:484 [inlined]
  [2] (::Pkg.Types.var"#52#55")(d::Dict{String, Any})
    @ Pkg.Types /Applications/Julia-1.9.app/Contents/Resources/julia/share/julia/stdlib/v1.9/Pkg/src/Types.jl:523
  [3] iterate
    @ ./generator.jl:47 [inlined]
  [4] _collect(c::Vector{Any}, itr::Base.Generator{Vector{Any}, Pkg.Types.var"#52#55"}, #unused#::Base.EltypeUnknown, isz::Base.HasShape{1})
    @ Base ./array.jl:802
  [5] collect_similar(cont::Vector{Any}, itr::Base.Generator{Vector{Any}, Pkg.Types.var"#52#55"})
    @ Base ./array.jl:711
  [6] map(f::Function, A::Vector{Any})
    @ Base ./abstractarray.jl:3261
  [7] (::Pkg.Types.var"#51#54"{String, Dates.DateTime, String})()
    @ Pkg.Types /Applications/Julia-1.9.app/Contents/Resources/julia/share/julia/stdlib/v1.9/Pkg/src/Types.jl:523
  [8] mkpidlock(f::Pkg.Types.var"#51#54"{String, Dates.DateTime, String}, at::String, pid::Int32; kwopts::Base.Pairs{Symbol, Int64, Tuple{Symbol}, NamedTuple{(:stale_age,), Tuple{Int64}}})
    @ FileWatching.Pidfile /Applications/Julia-1.9.app/Contents/Resources/julia/share/julia/stdlib/v1.9/FileWatching/src/pidfile.jl:82
  [9] mkpidlock
    @ /Applications/Julia-1.9.app/Contents/Resources/julia/share/julia/stdlib/v1.9/FileWatching/src/pidfile.jl:79 [inlined]
 [10] #mkpidlock#6
    @ /Applications/Julia-1.9.app/Contents/Resources/julia/share/julia/stdlib/v1.9/FileWatching/src/pidfile.jl:77 [inlined]
 [11] mkpidlock
    @ /Applications/Julia-1.9.app/Contents/Resources/julia/share/julia/stdlib/v1.9/FileWatching/src/pidfile.jl:77 [inlined]
 [12] write_env_usage(source_file::String, usage_filepath::String)
    @ Pkg.Types /Applications/Julia-1.9.app/Contents/Resources/julia/share/julia/stdlib/v1.9/Pkg/src/Types.jl:511
 [13] download_artifacts(env::Pkg.Types.EnvCache; platform::Base.BinaryPlatforms.Platform, julia_version::VersionNumber, verbose::Bool, io::Base.TTY)
    @ Pkg.Operations /Applications/Julia-1.9.app/Contents/Resources/julia/share/julia/stdlib/v1.9/Pkg/src/Operations.jl:759
 [14] add(ctx::Pkg.Types.Context, pkgs::Vector{Pkg.Types.PackageSpec}, new_git::Set{Base.UUID}; preserve::Pkg.Types.PreserveLevel, platform::Base.BinaryPlatforms.Platform)
    @ Pkg.Operations /Applications/Julia-1.9.app/Contents/Resources/julia/share/julia/stdlib/v1.9/Pkg/src/Operations.jl:1390
 [15] add
    @ /Applications/Julia-1.9.app/Contents/Resources/julia/share/julia/stdlib/v1.9/Pkg/src/Operations.jl:1372 [inlined]
 [16] add(ctx::Pkg.Types.Context, pkgs::Vector{Pkg.Types.PackageSpec}; preserve::Pkg.Types.PreserveLevel, platform::Base.BinaryPlatforms.Platform, kwargs::Base.Pairs{Symbol, Base.TTY, Tuple{Symbol}, NamedTuple{(:io,), Tuple{Base.TTY}}})
    @ Pkg.API /Applications/Julia-1.9.app/Contents/Resources/julia/share/julia/stdlib/v1.9/Pkg/src/API.jl:275
 [17] add(pkgs::Vector{Pkg.Types.PackageSpec}; io::Base.TTY, kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Pkg.API /Applications/Julia-1.9.app/Contents/Resources/julia/share/julia/stdlib/v1.9/Pkg/src/API.jl:156
 [18] add(pkgs::Vector{Pkg.Types.PackageSpec})
    @ Pkg.API /Applications/Julia-1.9.app/Contents/Resources/julia/share/julia/stdlib/v1.9/Pkg/src/API.jl:145
 [19] do_cmd!(command::Pkg.REPLMode.Command, repl::REPL.LineEditREPL)
    @ Pkg.REPLMode /Applications/Julia-1.9.app/Contents/Resources/julia/share/julia/stdlib/v1.9/Pkg/src/REPLMode/REPLMode.jl:412
 [20] do_cmd(repl::REPL.LineEditREPL, input::String; do_rethrow::Bool)
    @ Pkg.REPLMode /Applications/Julia-1.9.app/Contents/Resources/julia/share/julia/stdlib/v1.9/Pkg/src/REPLMode/REPLMode.jl:390
 [21] do_cmd
    @ /Applications/Julia-1.9.app/Contents/Resources/julia/share/julia/stdlib/v1.9/Pkg/src/REPLMode/REPLMode.jl:380 [inlined]
 [22] (::Pkg.REPLMode.var"#24#27"{REPL.LineEditREPL, REPL.LineEdit.Prompt})(s::REPL.LineEdit.MIState, buf::IOBuffer, ok::Bool)
    @ Pkg.REPLMode /Applications/Julia-1.9.app/Contents/Resources/julia/share/julia/stdlib/v1.9/Pkg/src/REPLMode/REPLMode.jl:557
 [23] #invokelatest#2
    @ ./essentials.jl:816 [inlined]
 [24] invokelatest
    @ ./essentials.jl:813 [inlined]
 [25] run_interface(terminal::REPL.Terminals.TextTerminal, m::REPL.LineEdit.ModalInterface, s::REPL.LineEdit.MIState)
    @ REPL.LineEdit /Applications/Julia-1.9.app/Contents/Resources/julia/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:2647
 [26] run_frontend(repl::REPL.LineEditREPL, backend::REPL.REPLBackendRef)
    @ REPL /Applications/Julia-1.9.app/Contents/Resources/julia/share/julia/stdlib/v1.9/REPL/src/REPL.jl:1300
 [27] (::REPL.var"#62#68"{REPL.LineEditREPL, REPL.REPLBackendRef})()
    @ REPL ./task.jl:514
```

**Comment by IanButterworth** [comment]
This should have been fixed in 1.9.1. A quick fix is deleting the usage files in the logs directory of your depot.

**Comment by liiyung** [comment]
Do you mean deleting the 3 files `artifact_usage.toml`, `manifest_usage.toml`, and `scratch_usage.toml`?

**Comment by IanButterworth** [comment]
Yeah. Or if you want to be more precise, look through them and fix any entries that are missing "time" fields by copying a time across from another. These files just affect when packages etc are tidied up by Pkg.

**Comment by liiyung** [comment]
I've added the "time" part to the missing sections, and it works now. Thanks so much for the help!

**Comment by IanButterworth** [comment]
It turns out I was mitaken and this didn't get backported for 1.9.1 https://github.com/JuliaLang/Pkg.jl/pull/3492

I'll leave this open for now, so that others can see the fix, and will close when it's released

**Comment by felixcremer** [comment]
I think, that this can be closed, because it got backported to 1.9 by now.
